retrieve_nearby_books.py

```diff
@@ -0,0 +1,176 @@
+#code to get json response for /v5/bibs/search
+import requests
+import base64
+import PIL
+import io
+from PIL import Image, ImageDraw, ImageFont
+from collections import defaultdict
+import textwrap
+from IPython.display import HTML
+from io import BytesIO
+import struct
+
+"""
+There are two steps for getting a response using sierra API
+        1) Authenticating with the Patron API
+        2) Making a Sierra API call
+"""
+def load_client_key():
+    #Step1
+    #Loading Client key and secret key from the file
+    f=open('C:\\uc\\uclib\\credentials.txt', 'r')
+    lines = f.readlines()
+
+    user=lines[0].replace('\n','')
+    pwd=lines[1].replace('\n','')
+    
+    #Combining the API key and secret into a single string separated by a colon:
+    comb = user+':'+pwd
+    return comb
+
+def generate_encoded_key(combined_clientkey_secretkey):
+    #Converting the combined key:secret string into a Base64 string
+    comb_enc = base64.b64encode(bytes(combined_clientkey_secretkey, 'utf-8')).decode("utf-8")
+    return comb_enc
+
+def authorize_api(combined_key_encrypted):
+    #Setting the Header name field to authorization.
+    #Setting the Header value field to Basic, and inserting the Base64-encoded credentials
+    headers = {
+        'authorization': "Basic "+combined_key_encrypted,
+        'content-type': "application/json",
+    }
+
+    #Set the method to POST
+    r_step1 = requests.post('https://uclid.uc.edu:443/iii/sierra-api/v5/token', headers=headers)
+    data = r_step1.json()
+
+    #Here we will store the access_token in a variable named myToken
+    myToken = data['access_token']
+    return myToken
+
+def call_api_search(my_accessToken):
+    #Step2
+    #We will use the captured access token to make our API call
+    #Setting the Header value field to Bearer, and inserting the access token
+    headers1 = {
+        'authorization': 'Bearer '+str(my_accessToken), 
+        'content-type': "application/json",
+    }
+
+    #Set the method to GET and capture the json response into a variable
+    r_step2 = requests.get('https://uclid.uc.edu:443/iii/sierra-api/v5/bibs/search?fields=id%2Ctitle%2Cauthor%2CvarFields&text=PS3616.U%3F%3F%3F%3F', headers=headers1)
+    json_string = r_step2.json()
+    return json_string
```

Comment:
Combine the 3 lines into 1:
Like this: `return requests.get(api_string, headers=headers1).json()`